tests/output.rs

```diff
@@ -159,12 +159,34 @@ fn assert_failure_output(fixture_dir_name: &str, command_and_output: &str) {
         }));
 }
 
+fn assert_non_match_output(fixture_dir_name: &str, command_and_output: &str) {
```

Comment:
`--help` prints to `stdout` (not `stderr`) but we don't want to do the matches-style "output sorting" so this basically mimics `assert_failure_output()` but looks at `stdout` instead of `stderr`